apps/asap-server/test/handlers/research-output/index-handler.test.ts

```diff
@@ -20,36 +23,336 @@ describe('Research Output index handler', () => {
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce(
+        researchOutputResponse,
+      );
+
+      await indexHandler(updateEvent(roID));
+      await indexHandler(createEvent(roID));
+
+      expect(algoliaIndexMock.deleteObject).not.toHaveBeenCalled();
+      expect(algoliaIndexMock.saveObject).toHaveBeenNthCalledWith(1, {
+        ...researchOutputResponse,
+        title: 'Title 1',
+        objectID: researchOutputResponse.id,
+      });
+      expect(algoliaIndexMock.saveObject).toHaveBeenNthCalledWith(2, {
+        ...researchOutputResponse,
+        objectID: researchOutputResponse.id,
+      });
+    });
+
+    test('receives the events created and unpublished in correct order', async () => {
+      const roID = 'ro-1234';
+      const createEv = createEvent(roID);
+      const unpublishedEv = unpublishedEvent(roID);
+
+      const researchOutputResponse = {
+        ...getResearchOutputResponse(),
+        id: roID,
+      };
+
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce(
+        researchOutputResponse,
+      );
+      researchOutputControllerMock.fetchById.mockRejectedValue(Boom.notFound());
+
+      await indexHandler(createEv);
+      await indexHandler(unpublishedEv);
+
+      expect(algoliaIndexMock.deleteObject).toHaveBeenCalledTimes(1);
+      expect(algoliaIndexMock.deleteObject).toHaveBeenCalledWith(
+        unpublishedEv.detail.payload.id,
+      );
+      expect(algoliaIndexMock.saveObject).toHaveBeenCalledTimes(1);
+      expect(algoliaIndexMock.saveObject).toHaveBeenCalledWith({
+        ...researchOutputResponse,
+        objectID: researchOutputResponse.id,
+      });
+    });
+
+    test('receives the reverse created and unpublished in correct order', async () => {
+      const roID = 'ro-1234';
+      const createEv = createEvent(roID);
+      const unpublishedEv = unpublishedEvent(roID);
+
+      researchOutputControllerMock.fetchById.mockRejectedValue(Boom.notFound());
+      researchOutputControllerMock.fetchById.mockRejectedValue(Boom.notFound());
+
+      await indexHandler(unpublishedEv);
+      await indexHandler(createEv);
+
+      expect(algoliaIndexMock.saveObject).not.toHaveBeenCalled();
+      expect(algoliaIndexMock.deleteObject).toHaveBeenCalledWith(
+        unpublishedEv.detail.payload.id,
+      );
+    });
+
+    test('receives the events created and deleted in correct order', async () => {
+      const roID = 'ro-1234';
+      const createEv = createEvent(roID);
+      const deleteEv = deleteEvent(roID);
+
+      const researchOutputResponse = {
+        ...getResearchOutputResponse(),
+        id: roID,
+      };
+
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce(
+        researchOutputResponse,
+      );
+      researchOutputControllerMock.fetchById.mockRejectedValue(Boom.notFound());
+
+      await indexHandler(createEv);
+      await indexHandler(deleteEv);
+
+      expect(algoliaIndexMock.deleteObject).toHaveBeenCalledTimes(1);
```

Comment:
FYI knowing that both times it will see a 404 it probably means that the `deleteObject` method will be called twice - and its fine - what we really care about is that at the end the object is deleted